src/platform/silabs/wifi/SiWx/ncp/sl_si91x_ncp_utility.h

```diff
@@ -23,20 +23,23 @@
  */
 #pragma once
 
-#include "FreeRTOS.h"
-#include "em_usart.h"
-#include "semphr.h"
+#include <platform/silabs/wifi/ncp/spi_multiplex.h>
+
 #include "silabs_utils.h"
-#include "sl_spidrv_exp_config.h"
 #include "sl_status.h"
-#include <platform/silabs/wifi/SiWx/ncp/sl_board_configuration.h>
-#include <platform/silabs/wifi/ncp/spi_multiplex.h>
 
 #define USART_INITSYNC_BAUDRATE 12500000
 
 #if SL_SPICTRL_MUX
-sl_status_t spi_board_init(void);
+/**
+ * @fn   sl_si91x_host_spi_multiplex_init(void)
+ * @brief Initialize the SPI multiplexing for the WiFi NCP and other devices.
+ *        This function is called during the initialization of the WiFi NCP.
+ * @return None
+ */
+sl_status_t sl_si91x_host_spi_multiplex_init(void);
 #endif // SL_SPICTRL_MUX
 
-extern uint32_t rx_ldma_channel;
-extern uint32_t tx_ldma_channel;
+// TODO: clean-up after SI91X-17602 is fixed

```

Comment:
NCP requires a bunch of SPI operations without interruptions thus `sl_si91x_host_spi_cs_assert` and `sl_si91x_host_spi_cs_deassert` APIs should typically be indicative of that sequence of operations.